File: view_based_projects/01-1-Counter/app/src/main/java/com/example/myapplication15/MainViewModel.java
package com.example.myapplication15;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainViewModel extends ViewModel {
    private MutableLiveData<Integer> counter = new MutableLiveData<>(0);

    public LiveData<Integer> getCounter() {
        return counter;
    }

    public void incrementCounter() {
        int newValue = counter.getValue() + 1;
        counter.setValue(newValue);
    }
}

File: 06-3-BoardOfMessageApp_kt2/app/src/main/java/com/example/boardofmessageapp_kt2/ui/compose/MainViewModel.kt
package com.example.boardofmessageapp_kt2.ui.compose

import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.getValue
import androidx.compose.runtime.setValue
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.boardofmessageapp_kt2.repository.common.BoardRepository
import kotlinx.coroutines.launch

class MainViewModel : ViewModel() {
    private val repository = BoardRepository()

    var numberOfBoardsMsg by mutableStateOf("")

    fun getBoards() {
        viewModelScope.launch {
            val boards = repository.getBoards()
            var msg = "#Boards: " + boards.size
            boards.forEach {
                msg += "\n" + it.name + ", messages: " + it.messages.size
            }

            numberOfBoardsMsg = msg
        }
        numberOfBoardsMsg = "loading..."
    }
}